### PR TITLE
[Feat](Authentication)Supporting Authentication Plugins and Kernel Isolation

### PR DESCRIPTION
**Pull request by CalvinKirs**
## Proposed changes 
- Use a Child-First ClassLoader to isolate plugins from the kernel, giving priority to plugin classes to prevent conflicts between the plugin and kernel classes. 
- Allow users to place plugins in a specified directory, such as auth-lib, to avoid classpath conflicts by default. 
- Support developers in debugging plugins by allowing them to bring in plugins via Maven, making the debugging process more convenient.

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/common/util/ClassLoaderUtils.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+public class ClassLoaderUtils {
+
+    private static final Map<String, String> pluginDirMapping = new HashedMap();
+
+    static {
+        pluginDirMapping.put(AuthenticatorFactory.class.getSimpleName(), "auth-lib");
```
Comment (CalvinKirs) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/ClassLoaderUtils.java:39]: This directory might not be ideal; perhaps we should use plugin/auth instead?

**Comment by CalvinKirs** [comment]
run buildall

**Comment by CalvinKirs** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by CalvinKirs** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Review by JNSimba — APPROVED** [review]

**Pull request merged**